sky/provision/fluidstack/instance.py

```diff
@@ -89,18 +89,47 @@ def run_instances(region: str, cluster_name_on_cloud: str,
 
     while True:
         instances = _filter_instances(cluster_name_on_cloud, pending_status)
+        if len(instances) > config.count:
+            raise RuntimeError(
+                f'Cluster {cluster_name_on_cloud} already has '
+                f'{len(instances)} nodes, but {config.count} are '
+                'required. Please try terminate the cluster and retry.')
         if not instances:
             break
         logger.info(f'Waiting for {len(instances)} instances to be ready.')
         time.sleep(POLL_INTERVAL)
     exist_instances = _filter_instances(cluster_name_on_cloud, ['running'])
     head_instance_id = _get_head_instance_id(exist_instances)
 
+    def rename(instance_id, new_name):
+        try:
+            utils.FluidstackClient().rename(instance_id, new_name)
+        except Exception as e:
+            logger.warning(f'run_instances error: {e}')
+            raise
+
+    for inst_id, inst in exist_instances.items():
+        if head_instance_id is None:
+            head_instance_id = inst_id
+            instance_name = f'{cluster_name_on_cloud}-head'
+            rename(inst_id, instance_name)
+            logger.info(f'Rename head node {head_instance_id} to '
+                        f'{instance_name}')
+        if inst_id != head_instance_id and inst['hostname'].endswith('-head'):
+            logger.info(f'Rename worker node {inst_id}.')
```

Comment:
nit: Be consistent in terms of logging + op ordering. "Renaming" here? L115-116 switch order?